File: fap_online_ui/lib/models/auth_response.dart
class AuthResponse {
  final String? token;
  final String? username;
  final String? fullName;
  final String? message;
  final String? role;

  AuthResponse({
    this.token,
    this.username,
    this.fullName,
    this.message,
    this.role,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      token: json['token'] as String?,
      username: json['username'] as String?,
      fullName: json['fullName'] as String?,
      message: json['message'] as String?,
      role: json['role'] as String?,
    );
  }
}

File: fap_online_backend/fap_online_backend/src/main/java/com/example/demo/dto/AuthResponse.java
package com.example.demo.dto;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AuthResponse {
    private Integer userId;
    private String token;
    private String username;
    private String fullName;
    private String role;
    private String message;
}
